client/src/components/desk/DeskContainer.tsx

```diff
@@ -1,48 +1,131 @@
+    const submitted = getRequestsWithStatus(requests, SUBMITTED);
+    const approved = getConsolidatedRequestsWithStatus(requests, APPROVED);
+    const readyForPickup = getConsolidatedRequestsWithStatus(requests, READY_FOR_PICKUP);
+
     return (
         <div>
-            <Header size="huge">Hardware Desk</Header>
+            <Header size="huge">
+                Hardware Desk
+                <Header.Subheader>{randomPhrase}</Header.Subheader>
+            </Header>
+
             <Grid stackable>
                 <Grid.Row columns={3}>
-                    <SubmittedList/>
-                    <CardList title="Ready to Prepare" length={0}>
-                        <Segment placeholder>
-                            <Container textAlign="center">
-                                <Header>
-                                    Nothing to prepare. Take a break!
-                                </Header>
-                            </Container>
-                        </Segment>
-                    </CardList>
-                    <CardList title="Ready for Pickup" length={0}>
-                        <Segment placeholder>
-                            <Container textAlign="center">
-                                <Header>
-                                    No requests awaiting pickup. Take a break!
-                                </Header>
-                            </Container>
-                        </Segment>
-                    </CardList>
+                    <SubmittedList loading={query.loading} requests={submitted} subscribeToUpdatedRequests={() => {
+                        subscribeToMore({
+                            document: REQUEST_CHANGE,
+                            updateQuery: (prev, {subscriptionData}) => {
+                                if (!subscriptionData.data) {
+                                    return prev;
+                                }
+                                console.log("Original requests array");
+                                prev.requests.forEach((x: any) => console.log(x));
+                                const updatedRequest = subscriptionData.data;
+                                const index = prev.requests.findIndex((x: any) => {
+                                    return x.request_id === updatedRequest.request_change.request_id;
+                                });
+
+                                console.log("Found request #", updatedRequest.request_change.request_id, " at index", index, ", new status is", updatedRequest.request_change.status);
+                                const requests = prev.requests;
+
+                                if (index === -1) { // request wasn't returned with original query; add it to array of requests
+                                    // If adding a new request to the array of requests, you have to do it this way rather than just pushing to the array
+                                    return Object.assign({}, prev, {
```

Comment:
So a lot of this is Apollo client subscriptions wizardry.

This function is updating the query results when new data comes in via the subscription.  When that happens we have to take the data from the subscription we just got and merge it into the cache.

`prev` contains the previous query results, which is just an object with one property (`requests`, from the name of the query).

There are 2 cases when merging in the subscription data, keeping in mind that Apollo will take what we return in this function and automatically normalize it into its internal cache:

1. The request is already being displayed and its data is being updated.  This is most likely a status change, but it could be any request property.  All we do is find the index of the request in the array and replace the value there with the new request.
2. The request was not previously on screen, so we have to add it to the `requests` array.  Pushing should work, intuitively, but when I used that the new requests didn't appear on screen.  It must be something with Apollo recognizing that the array has changed, because requests that weren't on screen and changed to a status where they should be on screen started appearing after I made the change to the current method.  The current method is also in line with the Apollo docs on this, so it's not like this is a hack in that sense.

Lastly, since prev only has the one property, normally you would have to be more careful, but in this case we can just return a new object with identical structure.